packages/vanilla/src/components/horizontalrule/horizontalrule.template.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Horizontal Rule template literal
+ *
+ * @param {object} props Property object passed into the template
+ * @param {string} props.style solid vs dashed [''|'dashed']
+ * @param {string} props.size size of the hr [''|'small'|'medium'|'large']
+ * @param {string} props.contrast color of the rule [''|'low'|'high']
+ * @param {string} props.weight weight of the rule [''|'thick']
+ * @returns {string} ES6 template literal of the horizontal rule
+ */
+const horizontalruleTemplate = ({ style, size, contrast, weight }) => {
+  const hrStyle = style === 'dashed' ? 'bx--hr--dashed' : '';
+  const hrSize = size && size !== '' ? `bx--hr--${size}` : '';
+  const hrContrast =
+    contrast && contrast !== '' ? `bx--hr--${contrast}-contrast` : '';
+  const hrWeight = weight === 'thick' ? 'bx--hr--thick' : '';
+
+  return `<hr
+    data-autoid="dds--hr"
+    class="bx--hr ${hrStyle} ${hrSize} ${hrContrast} ${hrWeight}"
```

Comment:
Does it make sense to use the `classnames` package here?

```suggestion
    class="${classnames('bx--hr', hrStyle, hrSize, hrContrast, hrWeight)}"
```

Reply:
I don't have a strong opinion either way, but in general I was hesitant to bring in any libraries for this, which is why I even opted against handlebars for the template rendering. 

Adding @annawen1 @kennylam @andysherman2121 @sannminwin for their opinion.